### PR TITLE
Fix crash when lambda uses non-interpolated string or empty array

### PR DESCRIPTION
**Pull request by samarsha**

### CODE REVIEW

File: src/QsCompiler/SyntaxProcessor/ContextVerification.fs [hunk 1]
```diff
@@ -365,7 +365,7 @@ let rec internal freeVariables e =
     | CONDITIONAL (e1, e2, e3) -> freeVariables e1 |> merge (freeVariables e2) |> merge (freeVariables e3)
     | ValueTuple es
     | StringLiteral (_, es)
-    | ValueArray es -> es |> Seq.map freeVariables |> Seq.reduce merge
+    | ValueArray es -> es |> Seq.map freeVariables |> Seq.fold merge Map.empty
```
Comment (ScottCarda-MS) [inline src/QsCompiler/SyntaxProcessor/ContextVerification.fs:368]: I can see how this fixes the case of an empty array, but I don't understand how this addresses the string case. Is there more to be done for this PR?

Comment (samarsha) [reply src/QsCompiler/SyntaxProcessor/ContextVerification.fs:368]: No, the PR is complete. This match arm handles all three cases: `ValueTuple es | StringLiteral (_, es) |  ValueArray es`.

Comment (ScottCarda-MS) [reply src/QsCompiler/SyntaxProcessor/ContextVerification.fs:368]: Oh I missed that, thanks. Was the original issue with strings something to do with empty strings `""`?

**Review by ScottCarda-MS — APPROVED** [review]

File: src/QsCompiler/SyntaxProcessor/ContextVerification.fs (hunk 1 above)
Comment (samarsha) [reply src/QsCompiler/SyntaxProcessor/ContextVerification.fs:368]: For `StringLiteral`, the second item `es` is the list of interpolated expressions, so the issue happens with both empty and non-empty strings if they are not interpolated.

Comment (ScottCarda-MS) [reply src/QsCompiler/SyntaxProcessor/ContextVerification.fs:368]: Ah okay, it all makes sense now, thanks!

**Pull request merged**